lib/plugin.c

```diff
@@ -377,12 +377,13 @@ static int load_plugins(char const *dir,
 
         int ret = 0;
 
-        if (plugins_to_load) {
+        if (plugins_to_load)
                 load_plugins_queue(dir, plugins_to_load);
-                (void) close(fd);
-        } else
+        else
                 ret = load_plugins_all(fd, dir);
 
+        (void) close(fd);
```

Comment:
> Shouldn't this be the way it was, only inside the if statement, since `closedir()` already closes the file descriptor and it is called after the for loop of the `load_plugins_all()` function?
> 
> Or is it to assure that even if `closedir()` is not successful the file descriptor is closed?

Yes, you are correct!  I incorrectly assumed that the `fdopendir()` call incremented the reference count on the file descriptor, but it actually claims it for its own use according to the [fdopendir(3)](https://linux.die.net/man/3/fdopendir) man page, which I find a bit counterintuitive.  I learned something new. :)

I've reverted the change, and added a comment about why `close()` is not called for the `load_plugins_all()` case.

Thanks for pointing this out!

Reply:
No problem.

> I've reverted the change, and added a comment about why `close()` is not called for the `load_plugins_all()` case.

This was actually one of the comments I wanted to add. It seemed a little counterintuitive without it.